shared/simplestreams/products.go

```diff
@@ -38,6 +64,34 @@ type Product struct {
 	Variant string `json:"variant,omitempty"`
 }
 
+// CanonicalProduct represents a single Canonical product inside download.json.
+type CanonicalProduct struct {
+	BaseProduct
+	LXDRequirements map[string]string `json:"lxd_requirements,omitempty"`
+}
+
+// GetBaseProduct returns the base product.
+func (p CanonicalProduct) GetBaseProduct() BaseProduct {
+	return p.BaseProduct
+}
```

Comment:
`GetBaseProduct` is implemented in both `CanonicalProduct` and `IncusProduct`. Could this function be moved to `BaseProduct` instead to avoid duplicates?

```go
func (p BaseProduct) GetBaseProduct() BaseProduct {
	return p
}
```

Reply:
yes it could